packages/@azure/servicebus/data-plane/lib/core/messageReceiver.ts

```diff
@@ -709,17 +709,6 @@ export class MessageReceiver extends LinkEntity {
       // Removes the link and its session if they are present in rhea's cache.
       await this._closeLink(this._receiver);
 
-      if (this.receiverType === ReceiverType.batching) {
-        log.error(
-          "[%s] Receiver '%s' with address '%s' is a Batching Receiver, so we will not be " +
-            "re-establishing the receiver link.",
-          connectionId,
-          this.name,
-          this.address
-        );
-        return;
-      }
```

Comment:
We still need the above. There are other code paths through which `detached()` can be called on a receiver.